CHANGELOG.md

```diff
@@ -35,6 +35,7 @@
 
 - Add `skip-get-modules` builder flag to support isolated environment executions (#6009)
   - Skip unnecessary Go binary path validation when the builder is used with `skip-compilation` and `skip-get-modules` flags (#6026)
+- Add mapstructure hook function for confmap.Unmarshaler interface (#6029)
```

Comment:
Enhancements or Breaking change? :)))

I don't think if I should worry about the possible breaking change... I would prefer to not :)

Reply:
I think this would only be breaking if a non-component struct already implements this interface. In that case, it would be possible that the behavior changes. However, I cannot find any examples of this in either core or contrib. The only implementation I find is otlpreceiver's Config, which will not have a behavior change because `UnmarshalReceiver` already accounts for component configs.